packages/hix/lib/Hix/Optparse.hs

```diff
@@ -40,16 +42,31 @@ pathOption desc parse =
   eitherReader \ raw ->
     first (const [exon|not a valid #{desc} path: #{raw}|]) (parse raw)
 
+absPathOrCwd ::
+  String ->
+  Path Abs Dir ->
+  (String -> Either e (SomeBase t)) ->
+  (String -> Either String (Path Abs t))
+absPathOrCwd desc cwd parse raw = 
+  first (const [exon|not a valid #{desc} path: #{raw}|]) (parse raw') <&> \case
+    Abs p -> p
+    Rel p -> foldr ($) cwd (replicate nbPrefixParents parent) </> p
+  where
+    raw' :: String
+    raw' = intercalate [pathSeparator] $ drop nbPrefixParents parts
+
+    nbPrefixParents :: Int
+    nbPrefixParents = length $ takeWhile (== "..") parts
```

Comment:
```suggestion
    Rel p -> foldr (const parent) cwd parents </> p
  where
    raw' :: String
    raw' = intercalate [pathSeparator] rest

    (parents, rest) = span (== "..") parts
```

I think it can be reduced to something like this!

Reply:
nice! I pushed it. Feel free to merge it if that's okay. I am working/experimenting on the delayed resolving in another PR